sycl/doc/extensions/experimental/sycl_ext_oneapi_uniform.asciidoc

```diff
@@ -98,8 +113,9 @@ diagnostic if it is known that an expression can never be uniform.
 Several classes in SYCL encapsulate information that is known to vary across
 work-items; instances of such classes can never be uniform. An implementation
 must issue a diagnostic if the `sycl::ext::oneapi::experimental::uniform` class template is
-instantiated with any of the following types: `sycl::item`, `sycl::nd_item`,
-`sycl::h_item`, `sycl::group`, `sycl::sub_group` and `sycl::nd_range`.
+instantiated with any of the following types: `sycl::id`, `sycl::item`,
```

Comment:
I don't agree with adding `sycl::id` to this list.

All the other disallowed types have members which return values relative to the calling work-item, and always represent properties of a specific work-item.  It doesn't make sense to declare such types as `uniform` because they can't be.  But users can construct arbitrary `sycl::id` objects, to represent arbitrary offsets, and it's possible for those user-constructed `id` objects to be uniform across work-items.

It wouldn't make sense for a user to declare the `id` they get back from something like `item::get_id()` as uniform, but I think that's already covered by other parts of the specification -- it's UB because the value is not actually uniform, not because it's an `id`.

Reply:
In that case, it seems like the implementation is wrong?  See:

https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/uniform.hpp#L60

Reply:
Yes, I think so.

Reply:
Restored the original spec wording in 64459035a1ef109fc48d539e976d658b610f76af.  That commit also fixes the implementation, aligning it with the spec.